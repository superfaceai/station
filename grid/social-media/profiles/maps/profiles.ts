/* eslint-disable jest/no-export */
import { SuperfaceTest } from '@superfaceai/testing';

type InputOptions = {
  profileIds?: string[];
  usernames?: string[];
};

export const profilesTest = (provider: string, input: InputOptions): void => {
  describe(`social-media/profiles/${provider}`, () => {
    let superfaceProfiles: SuperfaceTest;

    beforeEach(() => {
      superfaceProfiles = new SuperfaceTest({
        profile: 'social-media/profiles',
        provider,
      });
    });

    describe('GetProfiles', () => {
      describe('when access token is valid', () => {
        it('returns a list of profiles', async () => {
          await expect(
            superfaceProfiles.run({
              useCase: 'GetProfiles',
              input: {
                profileIds: input.profileIds,
              },
            })
          ).resolves.toMatchSnapshot();
        });
      });
    });

    describe('GetProfilesByUsername', () => {
      describe('when access token is valid', () => {
        it('returns a list of profiles', async () => {
          await expect(
            superfaceProfiles.run({
              useCase: 'GetProfilesByUsername',
              input: {
                usernames: input.usernames,
              },
            })
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
