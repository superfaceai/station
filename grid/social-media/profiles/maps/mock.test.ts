import { profilesTest } from './profiles';

profilesTest('mock', {
  profileIds: ['1232783237623119872'],
  usernames: ['NASAPersevere'],
});
