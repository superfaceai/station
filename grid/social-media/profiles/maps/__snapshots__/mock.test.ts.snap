// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social-media/profiles/mock GetProfiles when access token is valid returns a list of profiles 1`] = `
Ok {
  "value": Object {
    "profiles": Array [
      Object {
        "description": "NASA Mars rover. Launch: July 30, 2020. Landing: Feb. 18, 2021. Hobbies: Photography, collecting rocks, off-roading. ðŸš€ Team HQ @NASAJPL",
        "followersCount": 2909944,
        "followingCount": 58,
        "id": "1232783237623119872",
        "imageUrl": "https://pbs.twimg.com/profile_images/1379527227436531712/QasZP9s-_normal.jpg",
        "postsCount": 860,
        "profileUrl": "https://twitter.com/NASAPersevere",
        "username": "NASAPersevere",
        "website": "https://t.co/GpTOmL7zGl",
      },
    ],
    "rateLimit": Object {
      "bucket": "User lookup",
      "remainingRequests": 12,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1643713585,
      "totalRequests": 15,
    },
  },
}
`;

exports[`social-media/profiles/mock GetProfilesByUsername when access token is valid returns a list of profiles 1`] = `
Ok {
  "value": Object {
    "profiles": Array [
      Object {
        "description": "NASA Mars rover. Launch: July 30, 2020. Landing: Feb. 18, 2021. Hobbies: Photography, collecting rocks, off-roading. ðŸš€ Team HQ @NASAJPL",
        "followersCount": 2909944,
        "followingCount": 58,
        "id": "1232783237623119872",
        "imageUrl": "https://pbs.twimg.com/profile_images/1379527227436531712/QasZP9s-_normal.jpg",
        "postsCount": 860,
        "profileUrl": "https://twitter.com/NASAPersevere",
        "username": "NASAPersevere",
        "website": "https://t.co/GpTOmL7zGl",
      },
    ],
    "rateLimit": Object {
      "bucket": "User lookup",
      "remainingRequests": 12,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1643713585,
      "totalRequests": 15,
    },
  },
}
`;
